Homework3/LZW/LZW/LZW.cs

```diff
@@ -0,0 +1,150 @@
+
+        // Reading all bytes from a file
+        var stringToConvert = File.ReadAllBytes(pathToFileToCompress);
+
+        // Creating bor
+        Bor bor = new();
+
+        for (int i = 0; i < stringToConvert.Length; i++)
+        {
+            // If the byte is contained in the bor
+            if (bor.Contains(stringToConvert[i]))
+            {
+                // Moving on to the next byte
+                bor.MoveIntoDesiredNode(stringToConvert[i]);
+            }
+            // Otherwise, we add it to the bor
+            else
+            {
+                // Taking the idex from the parent vertex and encode it
+                var bytes = BitConverter.GetBytes(bor.GetCode());
+
+                // Cut off the extra bytes
+                // The required number of bytes is selected according to the size of the bor
+                // Since among the numbers that need to be encoded there may be a bor size (index of the maximum vertex),
+                // the number of bytes to store depends on the size of the bor.
+                // Even a number requires fewer bytes,
+                // it is written in a large number of bytes so that it can be decoded later.
+                Array.Resize(ref bytes, NumberOfBytes(bor.Size - 1));
+                bor.Add(stringToConvert[i]);
+                fs.Write(bytes);
+
+                // Going back to the last vertex
+                i--;
+            }
+        }
+
+        // The last bytes that are already in the dictionary and that are not written in the loop
+        var newbytes = BitConverter.GetBytes(bor.GetCode());
+        Array.Resize(ref newbytes, NumberOfBytes(bor.Size));
+        fs.Write(newbytes);
+    }
+
+    public static void DecompressFile(string pathToFile)
+    {
+
+        string fileName = Path.GetFileNameWithoutExtension(pathToFile);
+        fileName = $"{pathToFile}..//..//{fileName}";
+
+        // Create file
+        using FileStream fs = new(fileName, FileMode.Create);
+
+        // Reading all bytes from a file
+        byte[] stringToConvert = File.ReadAllBytes(pathToFile);
+
+        // Dictionary for decoding
+        var dictionary = new Dictionary<int, byte[]>();
+
+        for (int i = 0; i < 256; i++)
+        {
+            var byteArray = BitConverter.GetBytes(i);
+            dictionary.Add(i, byteArray[0..1]);
+        }
+
+        int rightBorder = 0;
+        int leftBorder = 0;
+        bool flag = false;
+
+        while (leftBorder < stringToConvert.Length)
+        {
+            // The right border of the current subarray
+            rightBorder = leftBorder + NumberOfBytes(dictionary.Count - 1) - 1;
+
+            // If the right border has gone beyond the edge, then we will make it the last byte
+            rightBorder = rightBorder > stringToConvert.Length - 1 ? stringToConvert.Length - 1 : rightBorder;
+
+            
+            if (leftBorder > stringToConvert.Length - 1)
```

Comment:
```suggestion
            rightBorder = rightBorder > stringToConvert.Length - 1 ? stringToConvert.Length - 1 : rightBorder;

            if (leftBorder > stringToConvert.Length - 1)
```
Две пустые строки подряд не принято